recaf-ui/src/main/java/me/coley/recaf/ui/pane/SearchPane.java

```diff
@@ -220,12 +210,113 @@ public static SearchPane createDeclarationSearch(String owner, String name, Stri
 		columns.add(modeLabel, comboMode);
 		columns.add(null, new ActionButton(Lang.getBinding("search.run"), () -> {
 			searchPane.searchDeclaration(
-					txtOwner.getText(), txtName.getText(), txtDesc.getText(),
-					comboMode.getValue());
+				txtOwner.getText(), txtName.getText(), txtDesc.getText(),
+				comboMode.getValue());
 		}));
 		return searchPane;
 	}
 
+	static class AdvancedSearch {
+		final ObjectProperty<Target> target = new SimpleObjectProperty<>();
+		final StringProperty identifier = new SimpleStringProperty();
+		final BooleanProperty deprecated = new SimpleBooleanProperty();
+		final BooleanProperty publiclyAccessible = new SimpleBooleanProperty();
+		final Member member = new Member();
+		final Type type = new Type();
+		final ObjectProperty<TargetType> targetType = new SimpleObjectProperty<>();
+		final BooleanProperty isStatic = new SimpleBooleanProperty();
+		final SetProperty<Where> where = new SimpleSetProperty<>();
+		SearchPane searchPane;
+
+
+		enum Target {
+			MEMBER, TYPE;
+		}
+
+		enum TargetType {
+			REFERENCE, DECLARATION;
+		}
+
+		enum Where {
+			IN_CODE, AS_RETURN_TYPE, AS_PARAMETER, AS_SUPER_TYPE
+		}
+
+		static class Member {
+
+			final ObjectProperty<Type> type = new SimpleObjectProperty<>();
+			final StringProperty returnType = new SimpleStringProperty();
+			final StringProperty parameters = new SimpleStringProperty();
+
+			enum Type {
+				FIELD, METHOD;
+			}
+
+			final StringProperty owner = new SimpleStringProperty();
+		}
+
+		static class Type {
+			final SetProperty<TypeType> type = new SimpleSetProperty<>();
+
+			enum TypeType {
+				CLASS, RECORD, ENUM, INTERFACE
+			}
+
+		}
+	}
+
+	public static SearchPane creatAdvancedSearch() {
+		StringBinding title = Lang.formatBy("%s: %s", Lang.getBinding("menu.search"),
+			Lang.getBinding("menu.search.advanced"));
+		var advancedSearch = new AdvancedSearch();
+		var whenTargetIsMember = advancedSearch.target.isEqualTo(AdvancedSearch.Target.MEMBER);
+		var whenTargetIsMethod = whenTargetIsMember
+			.and(advancedSearch.member.type.isEqualTo(AdvancedSearch.Member.Type.METHOD));
+		var whenTargetTypeIsDeclaration = advancedSearch.targetType.isEqualTo(AdvancedSearch.TargetType.DECLARATION);
+		var whenTargetIsType = advancedSearch.target.isEqualTo(AdvancedSearch.Target.TYPE);
+
+
+		return advancedSearch.searchPane = new SearchPane(title, Containers
+			.gridAsRows(2)
+			.padding(5, 10, 5, 10)
+			.columnConstraint(c -> c.percentWidth(30))
+			.columnConstraint(c -> c.fillWidth(true).h(HPos.LEFT, Priority.ALWAYS))
+			.vgap(5)
+			.add(label("Target"), choice(AdvancedSearch.Target.values(), AdvancedSearch.Target::name, AdvancedSearch.Target.MEMBER, SelectionMode.SINGLE)
+				.bind(advancedSearch.target))
+			.add(label("Identifier"), textField().bindText(advancedSearch.identifier))
+			.add(label("Deprecated"), checkbox().bind(advancedSearch.deprecated))
+			.add(label("Public"), checkbox().bind(advancedSearch.publiclyAccessible))
```

Comment:
Should we generally have a section for _"flags"_?

Would allow more fine-tuning of the search operations. I mentioned it in another comment, but flags could have 3-states (dont-care, active, inactive)

Then we could feed in a collection of states to check (access-modifiers, annotated, etc)